### PR TITLE
Ability to find keys which are NULL

### PR DESCRIPTION
**Pull request by Wouter0100**
When parent_id is NULL, we need to send it as "parent_id IS NULL", not "parent_id => NULL".

### CODE REVIEW

**Comment by rchavik** [comment]
Looks good.

Can you help adding tests? This is a regression due to syntax difference for null between cake 2.x -> 3.x.  Would like to catch these type of errors in the future.

File: src/Model/Table/AclNodesTable.php [hunk 1]
```diff
@@ -154,7 +154,7 @@ public function node($ref = null)
             foreach ($ref as $key => $val) {
                 if (strpos($key, $type) !== 0 && strpos($key, '.') === false) {
                     unset($ref[$key]);
-                    $ref["{$type}0.{$key}"] = $val;
+                    $ref[$type . '0.' . $key . ((is_null($val)) ? ' IS' : '')] = $val;
```
Comment (dakota) [inline src/Model/Table/AclNodesTable.php]: You can simply do `"{$type}0.{$key} IS"` as the ORM will automatically remove the `IS` if the value is not `null`

Comment (Wouter0100) [inline src/Model/Table/AclNodesTable.php]: Ah, awesome. Wasn't aware of that, nice. Is there any standard for these quotes, when we have to use a single or double quotes?

**Comment by Wouter0100** [comment]
Of course. I will add tests to this PR :)

**Comment by dakota** [comment]
@Wouter0100 Where are our tests :smile:

**Comment by Wouter0100** [comment]
They are in progress :grinning:.

**Comment by dakota** [comment]
:+1:

**Comment by Wouter0100** [comment]
I'm unable to reproduce the error at the moment.. I'll get back to this when I'm able to reproduce it, I hope this weekend :).

**Comment by dakota** [comment]
@Wouter0100 have you had a chance to reproduce this issue?

**Comment by Wouter0100** [comment]
I wasn't, unfortunately. I think it's a bug which is fixed in Cake or something..

**Comment by dakota** [comment]
Going to mark this as closed. If you do encounter it again feel free to open a new PR.
